src/lib/comms/include/sol-socket.h

```diff
@@ -154,26 +154,27 @@ struct sol_socket_type {
      * @brief Function to be called to read data from socket.
      *
      * @li @c s the socket pointer (this)
-     * @li @c buf the data buffer that will be used to receive the data
-     * @li @c len the size of @a buf
+     * @li @c buffer the data buffer that will be used to receive the data.
+     *     If @a buffer does not has the flags SOL_BUFFER_FLAGS_FIXED_CAPACITY or
+     *     SOL_BUFFER_FLAGS_MEMORY_NOT_OWNED this functions will allocate/re-allocate
+     *     the necessary memory. The @a buffer must be initialized.
      * @li @c cliaddr the source address of the message
      * @li returns @c the number of bytes received in success, otherwise a
      *     negative (errno) value is returned.
```

Comment:
the null bytes doesn't seem to be respected at the moment.

it doesn't have to do with `used`, but it does relate to `capacity`. If fixed capacity and there is a need for null-byte, that must be accounted in capacity.